### PR TITLE
Membership Refactor + CTA

### PR DESCRIPTION
**Pull request by FrederikBolding**
- Refactored membership detection + expiry (thanks @blurpesec )
- Added `MembershipPanel` with two states for now (either member or not)
- Added inverted button style

### CODE REVIEW

File: common/v2/features/Dashboard/components/MembershipPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import * as R from 'ramda';
+
+import { StoreContext } from 'v2/services';
+import { DashboardPanel, Typography, Button, Link } from 'v2/components';
+import { FONT_SIZE, COLORS, SPACING } from 'v2/theme';
+import { translateRaw } from 'v2/translations';
+import { ROUTE_PATHS } from 'v2/config';
+import defaultIcon from 'common/assets/images/membership/membership-none.svg';
+import { MEMBERSHIP_CONFIG } from 'v2/features/PurchaseMembership/config';
```
Comment (blurpesec) [inline common/v2/features/Dashboard/components/MembershipPanel.tsx]: tsk tsk

File: common/v2/services/Store/StoreProvider.tsx [hunk 2]
```diff
@@ -41,11 +41,16 @@ import { AccountContext, getDashboardAccounts } from './Account';
 import { SettingsContext } from './Settings';
 import { NetworkContext, getNetworkById } from './Network';
 import { findNextUnusedDefaultLabel, AddressBookContext } from './AddressBook';
+import { MembershipDict, MEMBERSHIP_CONFIG } from 'v2/features/PurchaseMembership/config';
+import { UnlockProtocolHandler } from '../EthService/network';
+import { DEFAULT_NETWORK } from 'v2/config';
```
Comment (blurpesec) [inline common/v2/services/Store/StoreProvider.tsx]: tsk tsk

File: common/v2/services/Store/BalanceService.tsx [hunk 3]
```diff
@@ -176,18 +178,28 @@ export const getAccountsTokenBalances = (accounts: StoreAccount[], tokenContract
...
+          R.toPairs(
+            R.mergeAll(
+              Object.keys(unlockStatusBalanceMap).map(address => ({
+                [address]: Object.keys(unlockStatusBalanceMap[address])
+                  .filter(b => unlockStatusBalanceMap[address][b].isGreaterThan(new BN(0)))
+                  .map(b => MEMBERSHIP_CONTRACTS[b])
+              }))
+            )
+          )
+        )
```
Comment (blurpesec) [inline common/v2/services/Store/BalanceService.tsx]: This whole thing is eye-cancer. Surely there has to be a better way to do this

File: common/v2/features/Dashboard/components/MembershipPanel.tsx (hunk 1 above)
Comment (blurpesec) [reply common/v2/features/Dashboard/components/MembershipPanel.tsx]: **nag** order imports **nag**

File: common/v2/services/Store/BalanceService.tsx (hunk 3 above)
Comment (FrederikBolding) [reply common/v2/services/Store/BalanceService.tsx]: I rewrote this in another way which seems cleaner 😄

**Pull request merged**